### PR TITLE
DASH-1074. Affiliate link page

### PR DESCRIPTION
**Pull request by andreyvEze**

### CODE REVIEW

File: client/src/hooks/refProfile.ts [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+  }, [refCode, dispatch]);
+
+  if (!refProfileInfo || refProfileInfo.type === REF_PROFILE_TYPE.REF)
+    return {};
+
+  return {
+    customHandleSubmit,
+    options,
+    defaultValue: options.length > 0 ? options[0] : undefined,
+    prefix: '@',
```
Comment (trukhilio) [inline client/src/hooks/refProfile.ts:222]: Use `FIO_ADDRESS_DELIMITER` constant

File: client/src/hooks/refProfile.ts [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+
+  useEffect(() => {
+    checkDomains(refProfileInfo?.settings?.domains);
+  }, [refProfileInfo?.settings?.domains, checkDomains]);
+
+  useEffect(() => {
+    dispatch(getDomains({ refCode }));
+  }, [refCode, dispatch]);
+  useEffect(() => {
+    if (refCode) dispatch(getSettings(refCode));
```
Comment (trukhilio) [inline client/src/hooks/refProfile.ts:212]: Remove  case actions.GET_REF_PROFILE_REQUEST:. If ok - don't need getSettings for ref.

File: client/src/components/GateVerificationComponent/GateVerificationComponent.tsx [hunk 3]
```diff
@@ -62,122 +56,106 @@ export const GateVerificationComponent: React.FC<Props> = props => {
...
-            </span>
-            . <br />
-            <span className={classes.boldText}>
-              Connect your Metamask wallet to complete validation and register
-              your FIO handle.
-            </span>
-          </p>
+      {isVerified
+        ? null
+        : !!onClick && (
```
Comment (trukhilio) [inline client/src/components/GateVerificationComponent/GateVerificationComponent.tsx:74]: Change back how it was

File: client/src/components/PurchaseNow/components/PurchaseEdgeWallet.tsx [hunk 4]
```diff
@@ -156,7 +156,7 @@ const PurchaseEdgeWallet: React.FC<Props> = props => {
                 .mul(DEFAULT_MAX_FEE_MULTIPLE_AMOUNT)
                 .round(0)
                 .toNumber(),
-              technologyProviderId: apis.fio.domainTpid,
+              technologyProviderId: apis.fio.affiliateTpid,
```
Comment (trukhilio) [inline client/src/components/PurchaseNow/components/PurchaseEdgeWallet.tsx:159]: Add ` apis.fio.affiliateTpid` to registerFioAddress

File: client/src/components/PurchaseNow/components/PurchaseLedgerWallet.tsx [hunk 5]
```diff
@@ -98,7 +98,7 @@ const PurchaseLedgerWallet: React.FC<Props> = props => {
             data = {
               fio_domain: registration.fioName,
               owner_fio_public_key: ownerFioPublicKey,
-              tpid: apis.fio.domainTpid,
+              tpid: apis.fio.affiliateTpid,
```
Comment (trukhilio) [inline client/src/components/PurchaseNow/components/PurchaseLedgerWallet.tsx:101]: Add ` apis.fio.affiliateTpid` to registerFioAddress

File: client/src/components/PurchaseNow/components/PurchaseMetamaskWallet.tsx [hunk 6]
```diff
@@ -169,7 +169,9 @@ export const PurchaseMetamaskWallet: React.FC<Props> = props => {
               owner_fio_public_key: ownerFioPublicKey,
               fio_address: registration.fioName,
               is_public: 0,
-              tpid: registration.isCombo ? apis.fio.domainTpid : apis.fio.tpid,
+              tpid: registration.isCombo
```
Comment (trukhilio) [inline client/src/components/PurchaseNow/components/PurchaseMetamaskWallet.tsx:172]: Add ` apis.fio.affiliateTpid` to registerFioAddress

File: client/src/hooks/refProfile.ts [hunk 7]
```diff
@@ -0,0 +1,236 @@
...
+        !!is_public &&
+        expiration &&
+        !isDomainExpired(fioDomain.name, expiration),
+    };
+  } catch (err) {
+    log.error(err);
+  }
+};
+
+export const useRefProfileAddressWidget = ({
```
Comment (trukhilio) [inline client/src/hooks/refProfile.ts:61]: Add comments what is going on here

File: client/src/pages/FioAffiliateProgramPage/FioAffiliateProgramPageContext.tsx [hunk 8]
```diff
@@ -67,13 +67,15 @@ export const useContext = (): FioAffiliateProgramPageContextProps => {
 
   const handleSelect = useCallback(
     (domainName: string) => {
-      const i = user?.affiliateProfile?.settings?.domains.findIndex(
+      // todo: add to local state
```
Comment (trukhilio) [inline client/src/pages/FioAffiliateProgramPage/FioAffiliateProgramPageContext.tsx:70]: Yes, do it in this PR

**Review by trukhilio — APPROVED** [review]
Check comments

File: client/src/components/PurchaseNow/components/PurchaseEdgeWallet.tsx (hunk 4 above)
Comment (andreyvEze) [reply client/src/components/PurchaseNow/components/PurchaseEdgeWallet.tsx:157]: done

File: client/src/components/PurchaseNow/components/PurchaseLedgerWallet.tsx (hunk 5 above)
Comment (andreyvEze) [reply client/src/components/PurchaseNow/components/PurchaseLedgerWallet.tsx:101]: done

File: client/src/hooks/refProfile.ts (hunk 7 above)
Comment (andreyvEze) [reply client/src/hooks/refProfile.ts:68]: added

File: client/src/hooks/refProfile.ts (hunk 2 above)
Comment (andreyvEze) [reply client/src/hooks/refProfile.ts]: let's discuss one more time

File: client/src/components/GateVerificationComponent/GateVerificationComponent.tsx (hunk 3 above)
Comment (andreyvEze) [reply client/src/components/GateVerificationComponent/GateVerificationComponent.tsx]: done

File: client/src/components/PurchaseNow/components/PurchaseMetamaskWallet.tsx (hunk 6 above)
Comment (andreyvEze) [reply client/src/components/PurchaseNow/components/PurchaseMetamaskWallet.tsx]: added

**Review by trukhilio — APPROVED** [review]

**Pull request merged**